x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -1203,6 +1204,26 @@ public void testCombineProjectionWithAggregationFirstAndAliasedGroupingUsedInAgg
         assertThat(Expressions.names(agg.groupings()), contains("first_name"));
     }
 
+    /**
+     * Expects
+     * Limit[1000[INTEGER]]
+     * \_Aggregate[STANDARD,[CATEGORIZE(first_name{f}#18) AS cat],[SUM(salary{f}#22,true[BOOLEAN]) AS s, cat{r}#10]]
+     *   \_EsRelation[test][_meta_field{f}#23, emp_no{f}#17, first_name{f}#18, ..]
+     */
+    public void testCombineProjectionWithCategorizeGrouping() {
+        var plan = plan("""
+            from test
+            | eval k = first_name, k1 = k
+            | stats s = sum(salary) by cat = CATEGORIZE(k)
+            | keep s, cat
+            """);
+
+        var limit = as(plan, Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        assertThat(Expressions.names(agg.aggregates()), contains("s", "cat"));
+        assertThat(Expressions.names(agg.groupings()), contains("cat"));
```

Comment:
I think the assertions here are insufficient because this would also pass if `CombineProjection` is not applied at all. IMO we should assert that
- before the `agg` is the `EsRelation`, and
- the `agg`'s groupings is not only named "cat", but also that it's an Alias that contains a Categorize that points to the FieldAttribute `first_name`.